planner/core/preprocess.go

```diff
@@ -992,6 +992,10 @@ func checkColumn(colDef *ast.ColumnDef) error {
 		if tp.Flen > mysql.MaxDecimalWidth {
 			return types.ErrTooBigPrecision.GenWithStackByArgs(tp.Flen, colDef.Name.Name.O, mysql.MaxDecimalWidth)
 		}
+
+		if tp.Decimal == 0 && tp.Flen == 0 {
```

Comment:
It would be better to add some comments for it.

Reply:
PTAL